### PR TITLE
Google auth

### LINKED ISSUE #22: Google Auth

### PR DESCRIPTION
**Pull request by rlwinter**
These changes can be reviewed in 3 sections: 
* foundational changes (Gemfile, db/schema, migration creating staff table), 
* implementing the rails scaffolded staffs resource (everything with staff in the name), 
* and changes to implement google_auth (omniauth, routes, sessions_controller, {login, welcome}.html.slim)

This closes #22

### CODE REVIEW

**Comment by bjmllr** [comment]
Do we have a way of switching off the external auth for testing?

File: spec/requests/pages_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe "Pages", type: :request do
+  describe "GET /" do
+    it "requesting the root of the website redirects to google to login" do
```
Comment (brkane) [inline spec/requests/pages_spec.rb]: Shouldn't a redirect be a 304?

**Comment by r33beers** [comment]
FYI, this needs a rebase

**Comment by brkane** [comment]
Needs big rebase

**Pull request closed without merging**